internal/lib/lib.go

```diff
@@ -190,30 +195,46 @@ func GetL7HttpRedirPolicy(vsName string) string {
 	return vsName
 }
 
-func GetSniNodeName(ingName, namespace, secret string, sniHostName ...string) string {
+func GetSniNodeName(ingName, namespace, secret, settingName string, sniHostName ...string) string {
+	namePrefix := NamePrefix
+	if settingName != "" {
+		namePrefix = NamePrefix + "-" + settingName + "-"
+	}
 	if len(sniHostName) > 0 {
-		return NamePrefix + sniHostName[0]
+		return namePrefix + sniHostName[0]
 	}
-	return NamePrefix + ingName + "-" + namespace + "-" + secret
+	return namePrefix + ingName + "-" + namespace + "-" + secret
 }
 
-func GetSniPoolName(ingName, namespace, host, path string, args ...string) string {
+func GetSniPoolName(ingName, namespace, host, path, infrasetting string, args ...string) string {
```

Comment:
Can't the infraSetting be an optional parameter as well? Since the args ...string is a slice. This way you don't have to touch code where you don't want to specify the infraSetting param. You can assume that args[0] is infraSetting?